### ISSUE TITLE
Large language model(LLM)

### ISSUE THREAD

**Issue by Koteeshwari** [Koteeshwari/Google-Generative-AI#2]

**Comment by Koteeshwari** [comment]
**Introduction to large language models**

<img width="430" height="268" alt="Image" src="https://github.com/user-attachments/assets/d20ae5e0-9893-4dca-b4d2-e1b0e883fc83" />

*Large Language Models, or LLMs, are a subset of deep learning.

<img width="222" height="157" alt="Image" src="https://github.com/user-attachments/assets/3fc3c3a9-e23c-43e1-b9d1-a05af6042b31" />

<img width="207" height="321" alt="Image" src="https://github.com/user-attachments/assets/b93f527c-a025-4931-b54d-03b7c2646763" />

** generative AI**

This is a type of artificial intelligence that can produce new content - including text, images, audio, and synthetic data.

**what are Large Language Models?**

<img width="803" height="204" alt="Image" src="https://github.com/user-attachments/assets/d5ce4db7-44e1-47ed-907d-86fbae2d1cef" />

Large Language Models refer to large, general-purpose language models that can be pre-trained and then fine-tuned for specific purposes.

**What do pre-trained and fine-tuned mean?**
Imagine training a dog.
Often you train your dog basic commands such as sit, come, down, and stay.
These commands are normally sufficient for everyday life and help your dog become a good canine citizen.

*These models are trained for general purposes to solve common language problems such as text classification, question answering, document summarization, and text generation across industries.

<img width="788" height="417" alt="Image" src="https://github.com/user-attachments/assets/0b3a5c63-1683-4e2a-bbe4-12694e774d35" />

*The models can then be tailored to solve specific problems in different fields such as retail, finance, and entertainment, using a relatively small size of field datasets.

<img width="444" height="425" alt="Image" src="https://github.com/user-attachments/assets/d42e1945-9dba-4edc-8caf-bcd8fd18feb4" />

**How about letting them create fundamental language models for others to use?**
So this leaves us with our last terms: pre-trained and fine-tuned, which mean to pre-train a Large Language Model for a general purpose with a large dataset, and then fine-tune it for specific aims with a much smaller dataset.

**LLM use cases**
*First, a single model can be used for different tasks. This is a dream come true. These Large Language Models, that are trained with petabytes of data and generate billions of parameters, are smart enough to solve different tasks, including language translation, sentence completion, text classification, question answering, and more.
*Second, Large Language Models require minimal field training data when you tailor them to solve your specific problem. Large Language Models obtain decent performance even with little domain training data.
*Third, the performance of Large Language Models is continuously growing when you add more data and parameters. Large Language Models are almost exclusively based on transformer models.

<img width="860" height="315" alt="Image" src="https://github.com/user-attachments/assets/b9b526cf-5c95-42fc-a82e-4c03551f44a0" />


<img width="794" height="409" alt="Image" src="https://github.com/user-attachments/assets/a9eee626-7c37-45f0-aeb5-2dca6ff4b674" />

<img width="797" height="432" alt="Image" src="https://github.com/user-attachments/assets/3a2fe888-6475-4350-80b7-4c80c3b329fd" />


<img width="427" height="412" alt="Image" src="https://github.com/user-attachments/assets/d13dac90-c686-4fc6-bb0d-2c6a7c7995ba" />

<img width="833" height="420" alt="Image" src="https://github.com/user-attachments/assets/4fb664d3-9a63-4626-8fe5-bdede2bfc50d" />

<img width="430" height="200" alt="Image" src="https://github.com/user-attachments/assets/70d27fb9-ad6d-4654-9165-3fc3fb88471a" />

**Prompt design and prompt engineering are two closely related concepts in Natural Language Processing.**

Both involve the process of creating a prompt that is clear, concise, and informative.
**But, there are some key differences between the two.**
Prompt design is the process of creating a prompt that is tailored to the specific task that the system is being asked to perform.

<img width="402" height="317" alt="Image" src="https://github.com/user-attachments/assets/100c6a70-692d-4288-aab5-43689085bccf" />

*Prompt design is essential, while prompt engineering is only necessary for systems that require a high degree of accuracy or performance.
*There are three kinds of Large Language Models: generic language models, instruction-tuned, and dialog-tuned.

Each needs prompting in a different way.
*Generic language models predict the next word based on the language in the training data.

<img width="390" height="198" alt="Image" src="https://github.com/user-attachments/assets/49faa541-c5d5-471f-8977-02d3bde21528" />

<img width="1863" height="935" alt="Image" src="https://github.com/user-attachments/assets/c6ca9845-ac06-4324-901e-c59a3c866baa" />

<img width="346" height="148" alt="Image" src="https://github.com/user-attachments/assets/17f50ab4-f675-475b-9ef0-9f47160b0961" />

Example :-

<img width="389" height="375" alt="Image" src="https://github.com/user-attachments/assets/1bac2aad-e5ed-4d10-9ead-f9ffe31127e0" />

<img width="286" height="109" alt="Image" src="https://github.com/user-attachments/assets/333f73f3-cf6e-4ae5-be56-55274400cbed" />

<img width="363" height="305" alt="Image" src="https://github.com/user-attachments/assets/433adc09-d14f-46f6-8bbd-be2f6f11151d" />

<img width="429" height="214" alt="Image" src="https://github.com/user-attachments/assets/92b17f92-1fa8-45e5-ba2d-f44ca0a0c679" />

<img width="420" height="240" alt="Image" src="https://github.com/user-attachments/assets/3f3dcf62-0891-4177-95f2-6f585d6fb24a" />

<img width="412" height="402" alt="Image" src="https://github.com/user-attachments/assets/bf35b021-0714-418a-b499-03e33f5d38c8" />

<img width="381" height="430" alt="Image" src="https://github.com/user-attachments/assets/7b495452-b251-465c-acf8-5b3fa25ffaf5" />

<img width="407" height="429" alt="Image" src="https://github.com/user-attachments/assets/3979fd8b-a382-491c-b69f-dd4558c3a65a" />

<img width="310" height="366" alt="Image" src="https://github.com/user-attachments/assets/a84a926f-19c4-4f80-9433-36d05b3f5101" />

<img width="856" height="396" alt="Image" src="https://github.com/user-attachments/assets/8294c1e5-ddaa-41bd-ac97-09339ec31a54" />

**Comment by Koteeshwari** [comment]
<img width="395" height="393" alt="Image" src="https://github.com/user-attachments/assets/d4608125-30f2-4b46-a539-df6abdf4ad24" />

<img width="506" height="224" alt="Image" src="https://github.com/user-attachments/assets/4f43c95c-5694-4e9a-b1c6-317ed70841ee" />

<img width="340" height="399" alt="Image" src="https://github.com/user-attachments/assets/9c521a8d-a128-4de6-bec4-d2171b25fc3e" />

**Vertex AI can help you create your own: chat bots, digital assistants, custom search engines, knowledge bases, training applications, and more.**
<img width="257" height="420" alt="Image" src="https://github.com/user-attachments/assets/ee83eff2-dff3-46cd-bd12-27f6b478baa1" />
